rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/ClassFileListener.java

```diff
@@ -0,0 +1,201 @@
+                }
+
+                fileModel.asVertex().setProperty(JavaClassFileModel.SKIP_DECOMPILATION, true);
+            }
+            catch (IOException e)
+            {
+                LOG.log(Level.WARNING,
+                            "ASM was unable to parse class file: " + fileModel.getFilePath() + " due to: " + e.getMessage(),
+                            e);
+                ClassificationService classificationService = new ClassificationService(event.getGraphContext());
+                classificationService.attachClassification(context, fileModel, JavaClassFileModel.UNPARSEABLE_CLASS_CLASSIFICATION,
+                            JavaClassFileModel.UNPARSEABLE_CLASS_DESCRIPTION);
+                return;
+            }
+        }
+        finally
+        {
+            ExecutionStatistics.get().end("ClassFileListener.fileModelCreated()");
+        }
+    }
+
+    private void addClassFileMetadata(GraphRewrite event, EvaluationContext context, FileModel fileModel)
+    {
+        try (FileInputStream fis = new FileInputStream(fileModel.getFilePath()))
+        {
+            final ClassParser parser = new ClassParser(fis, fileModel.getFilePath());
+            final JavaClass bcelJavaClass = parser.parse();
+            final String packageName = bcelJavaClass.getPackageName();
+
+            final String qualifiedName = bcelJavaClass.getClassName();
+
+            final JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
+            final JavaClassModel javaClassModel = javaClassService.create(qualifiedName);
+            int majorVersion = bcelJavaClass.getMajor();
+            int minorVersion = bcelJavaClass.getMinor();
+
+            String simpleName = qualifiedName;
+            if (packageName != null && !packageName.equals("") && simpleName != null)
+            {
+                simpleName = StringUtils.substringAfterLast(simpleName, ".");
+            }
+
+            JavaClassFileModel javaClassFileModel = new GraphService<>(event.getGraphContext(), JavaClassFileModel.class).addTypeToModel(fileModel);
+            javaClassFileModel.setMajorVersion(majorVersion);
+            javaClassFileModel.setMinorVersion(minorVersion);
+            javaClassFileModel.setPackageName(packageName);
+
+            javaClassModel.setSimpleName(simpleName);
+            javaClassModel.setPackageName(packageName);
+            javaClassModel.setQualifiedName(qualifiedName);
+            javaClassModel.setClassFile(javaClassFileModel);
+            javaClassModel.setPublic(bcelJavaClass.isPublic());
+            javaClassModel.setInterface(bcelJavaClass.isInterface());
+
+            final String[] interfaceNames = bcelJavaClass.getInterfaceNames();
+            if (interfaceNames != null)
+            {
+                for (final String interfaceName : interfaceNames)
+                {
+                    JavaClassModel interfaceModel = javaClassService.getOrCreatePhantom(interfaceName);
```

Comment:
will this phatnom be converted into normal JavaClassFileModel/JavaSourceFileModel?